libbeat/publisher/pipeline/consumer.go

```diff
@@ -154,7 +154,7 @@ func (c *eventConsumer) loop(consumer queue.Consumer) {
 		}
 
 		paused = c.paused()
-		if !paused && c.out != nil && batch != nil {
+		if c.out != nil && batch != nil {
```

Comment:
This turned out to be a key change in getting the test to pass.

Consider the case where the consumer is paused (so `paused == true` here) but it still has a batch of events that it has consumed from the queue but not yet sent to the workqueue. In this scenario because of the `!paused` clause here, we would fall into the `else` clause, which would set `out = nil`. This would then cause the last `select` in this method to block sometimes (depending on timing). 

Now, if we have a valid output group `c.out` and a `batch`, we don't pay attention to the `paused` state; this ensures that this batch will be send to that output group's work queue via that final `select`'s final `case`.

Reply:
Oh wow nice fix!